src/providers/BackgroundBlurProvider/index.tsx

```diff
@@ -0,0 +1,133 @@
+    device: Device
+  ) => Promise<DefaultVideoTransformDevice>;
+  isBackgroundBlurSupported: boolean | undefined;
+}
+
+const BackgroundBlurProviderContext =
+  createContext<BackgroundBlurProviderState | undefined>(undefined);
+
+const BackgroundBlurProvider: FC<Props> = ({ spec, options, children }) => {
+  const [isBackgroundBlurSupported, setIsBackgroundBlurSupported] = useState<boolean | undefined>(undefined);
+  const [processor, setProcessor] = useState<VideoFrameProcessor | undefined>();
+
+  useEffect(() => {
+    async function initializeBackgroundBlur() {
+      try {
+        console.log('Initializing background blur processor.');
+        const createdProcessor = await BackgroundBlurVideoFrameProcessor.create(
+          spec,
+          options
+        );
+        // BackgroundBlurVideoFrameProcessor.create will return a NoOpVideoFrameProcessor
+        // in the case that BackgroundBlurVideoFrameProcessor.isSupported() returns false.
+        // BackgroundBlurVideoFrameProcessor.create() can also throw an error in case loading
+        // the assets are not fetched successfully.
+        if (createdProcessor instanceof NoOpVideoFrameProcessor) {
+          console.log(`Initialized NoOpVideoFrameProcessor.`);
+          setProcessor(undefined);
+          setIsBackgroundBlurSupported(false);
+        } else {
+          console.log(`Initialized background blur processor: ${JSON.stringify(processor)}`);
+          setProcessor(createdProcessor);
+          setIsBackgroundBlurSupported(true);
+        }
+      } catch (error) {
+        console.error(`Error creating a background blur video frame processor device.`, error);
+        setProcessor(undefined);
+        setIsBackgroundBlurSupported(false);
+      }
+    }
+    initializeBackgroundBlur();
+    return () => {
+      console.log(`Destroying background blur processor.`);
+      processor?.destroy();
+    };
+  }, []);
+
+  const createBackgroundBlurDevice = async (
+    selectedDevice: Device
+  ): Promise<DefaultVideoTransformDevice> => {
+    console.log(`Calling createBackgroundBlurDevice with device: ${selectedDevice}`);
+    // It takes time for the processor to start - if you call this method before it is finished initializing, throw an error
+    if (!isBackgroundBlurSupported) {
+      throw new Error('Background blur is not supported. The processor may not be initialized yet.');
+    }
+    try {
+      const logger = options?.logger
+        ? options.logger
+        : new ConsoleLogger('BackgroundBlurProvider', LogLevel.INFO);
+      if (processor) {
+        const chosenVideoTransformDevice = new DefaultVideoTransformDevice(
+          logger,
+          selectedDevice,
+          [processor]
+        );
+        console.log(`Created video transform device ${JSON.stringify(chosenVideoTransformDevice, null, 2)}`);
+        return chosenVideoTransformDevice;
+      } else {
+        throw new Error('Processor has not been created.');
```

Comment:
[For learning] We throw errors, when the processor is not created yet or will create in a while, is there a recommendation we can add in documentation on how long to wait or handle this creation with a retry, could you please provide some information on that?

Reply:
sure, will do.